cluster/kube-up.sh

```diff
@@ -32,6 +32,31 @@ fi
 
 source "${KUBE_ROOT}/cluster/kube-util.sh"
 
+DEPRECATED_PROVIDERS=(
+  "centos"
+  "libvert-coreos"
+  "local"
+  "openstack-heat"
+  "photon-controller"
+  "vagrant"
+  "vsphere"
+  "windows"
+)
+
+for provider in "${DEPRECATED_PROVIDERS[@]}"; do
+  if [[ "${KUBERNETES_PROVIDER}" == "${provider}" ]]; then
+    cat <<EOF 1>&2
+
+!!! DEPRECATION NOTICE !!!
+
+The '${provider}' kube-up provider is deprecated and will be removed in a future
```

Comment:
What @pipejakob 
Also we could state the timeline up front here (removal in v1.9?)